src/MetadataUtility/Metadata/SupportFiles/FrontierLabs/LogFile.cs

```diff
@@ -39,146 +50,339 @@ public static void FindLogFile(TargetInformation information, IEnumerable<string
+
+            return new Location() with
+            {
+                Latitude = latitude,
+                Longitude = longitude,
+            };
+        }
+
+        /// <summary>
+        /// Parses battery data from a log file.
+        /// </summary>
+        /// <param name="value">The unparsed battery data.</param>
+        /// <returns>
+        /// Parsed battery data.
+        /// </returns>
+        public static (double? BatteryLevel, double? Voltage) BatteryParser(string value)
+        {
+            string[] batteryValues = value.Split(" ", StringSplitOptions.RemoveEmptyEntries);
+
+            double? batteryLevel = double.Parse(NumericParser(batteryValues[0])) / 100;
+            double? batteryVoltage = double.Parse(NumericParser(batteryValues[1]));
+
+            return (batteryLevel, batteryVoltage);
+        }
+
+        /// <summary>
+        /// Parses a microphone from a log file.
+        /// </summary>
+        /// <param name="value">The unparsed microphone value.</param>
+        /// <returns>
+        /// A parsed microphone object.
+        /// </returns>
+        public static Microphone MicrophoneParser(string value)
+        {
+            if (value.Contains("Unknown"))
+            {
+                return null;
+            }
+
+            char? channelName = null;
+            int? channel = null;
+
+            // Parse channel data if it exists
+            if (value.Contains("Ch"))
+            {
+                channelName = value.Split(':').First().ToArray().Last();
+                channel = (int)channelName - 64;
             }
 
-            return logFile;
+            // Split string into individual microphone values
+            string[] micValues = value.Split("\"").Select(x => x.Trim()).ToArray();
+
+            // Parse each value
+            string uid = micValues[0].Split(" ").Last();
+            string type = micValues[1];
+            string buildDate = string.Join(
+                '-',
+                new string(micValues[2].Where(c => !(new char[] { ' ', '(', ')' }).Contains(c)).ToArray())
+                .Split('/').Reverse().ToArray());
+
+            return new Microphone() with
+            {
+                Channel = channel,
+                ChannelName = channelName?.ToString(),
+                UID = uid,
+                Type = type,
+                BuildDate = buildDate,
+            };
         }
 
-        public override void ExtractInformation()
+        /// <summary>
+        /// Helps parse numeric values by removing all non numeric characters.
+        /// </summary>
+        /// <param name="value">The unparsed numeric value.</param>
+        /// <returns>
+        /// A string containing only numeric characters.
+        /// </returns>
+        public static string NumericParser(string value) => new string(value.Where(c => char.IsDigit(c) || (new char[] { '.', '-', '+' }).Contains(c)).ToArray());
```

Comment:
A regular expression would be way more efficient here.

Something like: `[^0-9+-.]` (which matches any non-digit/period/minus/plus character) with the replace function on a regex, replacing any match with `''`